### ISSUE TITLE
how to run this ion my own datdaset 

### ISSUE THREAD

**Issue by YangBai1109** [Hsuxu/Magic-VNet#1]
i have a dataset about ABVS . how can i run this code on my dataset?

thank you very much

**Comment by cruigo93** [comment]
@PanPan0210 have you done it?

**Comment by cruigo93** [comment]
because I have just one class and when I make VNet(out_channels=1) loss does not change, but when I make background as separate class in separate channel it appears to work but not so great

**Comment by Hsuxu** [comment]
Delete the `softmax` function and retry.
https://github.com/Hsuxu/Magic-VNet/blob/4604b040386f527facbe2a9927fcc690b2854e62/magic_vnet/blocks.py#L208

**Comment by cruigo93** [comment]
thank you for reply! I use vnet with out_channels=1 and dice loss as follows 
```
def dice_loss(input, target):
    smooth = 1.
    loss = 0.
    # print(torch.unique(input), torch.unique(target))
    iflat = input.view(-1) 
    tflat = target.view(-1)
    intersection = (iflat * tflat).sum()
    
    loss += (1 - ((2. * intersection + smooth) /
                        (iflat.sum() + tflat.sum() + smooth)))
    return loss
```
but still i have issues, because loss is fluctuating. Can you help with that? maybe provide some demo code. Thank you

**Comment by Hsuxu** [comment]
You need applt `sigmoid` function in your loss function. Or you can use my implemention of [BinaryDiceLoss](https://github.com/Hsuxu/Loss_ToolBox-PyTorch/blob/master/DiceLoss/dice_loss.py)

**Comment by cruigo93** [comment]
```
    model = VNet(in_channels=1, out_channels=1)
    model = model.cuda()
    optimizer = optim.Adam(model.parameters(), lr=1e-4)
    nTrain = len(train_loader.dataset)
    nVal = len(val_loader.dataset)
    for epoch in range(50):
        model.train()
        nProcessed = 0
        for batch_idx, (data, target) in enumerate(train_loader):
            if torch.cuda.is_available():
                data, target = data.cuda(), target.cuda()
            optimizer.zero_grad()
            output = model(data.float())
            loss = criterion(output, target.float())
            loss.backward()
            optimizer.step()
            nProcessed += len(data)
            
            print('Train Epoch {}: [{}/{} ({:.0f}%)]\tLoss: {:.8f}'.format(
                epoch, nProcessed, nTrain, 100. * batch_idx / len(train_loader),
                loss.item()))
        model.eval()
        nProcessed = 0
        for batch_idx, (data, target) in enumerate(val_loader):
            if torch.cuda.is_available():
                data, target = data.cuda(), target.cuda()
            optimizer.zero_grad()
            # print(data.shape)
            output = model(data.float())
            loss = criterion(output, target.float())
            # make_graph.save('/tmp/t.dot', loss.creator); assert(False)
            
            nProcessed += len(data)
          
            print('Val Epoch {}:  [{}/{} ({:.0f}%)]\tLoss: {:.8f}'.format(
                epoch, nProcessed, nVal, 100. * batch_idx / len(train_loader),
                loss.item()))
    torch.save(model, './best_model.pth')
    print('Model saved!')
```

**Comment by cruigo93** [comment]
can you take a look? is it right?

**Comment by cruigo93** [comment]
because i have just empty masks

**Comment by Hsuxu** [comment]
The target should be a binary mask, and apply `sigmoid` function on output before compute the dice coefficient, like `loss = criterion(torch.sigmoid(output), target.float())`
